### DISCUSSION TITLE
How to Rebase Branches only during scheduled time

### DISCUSSION THREAD

**Discussion by msimperi** [renovatebot/renovate#13083] [Request Help] [answered]
Hello!

Is there way to schedule rebasing so that it would only happen when also new PRs can be created for the dependencies?

I have tried to schedule that normal updates would be rolled after work during Friday, but allow internal dependencies to be updated ASAP. Now I have seen that also rebase for the PRs updating non-internal dependencies gets rebased/updated often and by running also quality checks it adds lot of unnecessary builds to be happen.

Here is our config currently (some parts cleaned):

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "enabled": true,
  "extends": [
    "config:base",
    ":pinAllExceptPeerDependencies",
    ":semanticPrefixChore"
  ],
  "rebaseWhen": "behind-base-branch",
  "schedule": ["after 5pm on friday"],
  "bumpVersion": "patch",
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true
    },
    {
      "updateTypes": ["pin"],
      "schedule": ["at any time"]
    },
    {
      "packagePatterns": ["^@internal-packages"],
      "prPriority": 100,
      "schedule": ["at any time"],
      "enabled": true
    }
  ],
  "encrypted": {
    "npmToken": "<token>"
  },
  "npmrc": "<content>"
}
```

Here is a log from one update that checked schedule, but still rebased the branch:

![image](https://user-images.githubusercontent.com/2699541/145796478-ece1294c-2ec0-42b1-8b78-941547fd9798.png)

**Comment by rarkins** [comment, marked as answer]
https://docs.renovatebot.com/configuration-options/#updatenotscheduled

**Reply by msimperi** [reply]
Thanks for the ultimate fast help. I was trying to read everything, but somehow didn't see this. Thanks Rhys!
